external-crates/move/crates/move-package-alt/src/git/cache.rs

```diff
@@ -377,6 +365,40 @@ fn display_cmd(cmd: &Command) -> String {
     result
 }
 
+/// Tries to convert a "branch" or "tag" into a full SHA, using `ls-remote`.
+/// It first tries to find a tag, and if that fails, it fallbacks into a branch lookup.
+///
+/// Note: a "no results" search for tags or branches returns an empty result (`Ok("")`)
```

Comment:
I would make the second half of this doc comment internal comments since they're not relevant to the user of the function